Relocate ELFactoryWrapperForCDI

volosied opened the pull request:
### Summary of Change
_ELFactoryWrapperForCDI_ is  located in three bundles (each EL API), so I've deleted them and moved the file into a new bundle: `io.openliberty.el.internal.cdi`.   The APIs are now a bit cleaner. 

This new bundle is placed in the private EL features, which are the same features where this interfaces existed before. 

I also renamed the package `com.ibm.wsspi.el`  from to `io.openliberty.el.internal.cdi` as this interface is not an spi. 

The package rename also required a change in closeld liberty. The `com.ibm.ws.openwebbeans.impl` project uses _ELFactoryWrapperForCDI_.  The CL PR is  #24660


### Other Notes & Comments
Originally tried to place this interface in `com.ibm.ws.jsp` but that identified bundle resolution errors. Therefore this new bundle approach was chosen instead. 
See comments in #17223 for more details. 
Also see https://github.com/OpenLiberty/open-liberty/pull/18130#issuecomment-897784727

volosied committed:
Remove ELFactoryWrapperForCDI.java from EL APIs 2.2, 3.0, & 4.0

volosied commented on dev/com.ibm.ws.jsp/src/com/ibm/wsspi/el/ELFactoryWrapperForCDI.java:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2012 IBM Corporation and others.
+ * Copyright (c) 2021 IBM Corporation and others.
```
This is the new file in `com.ibm.ws.jsp` so it have it a 2021 copyright? Should it be 2012, 2021 instead?

tevans78 replied:
actually, if you are only moving it, then you shouldn't change the date at all

volosied replied:
I'll change it back. Thanks

volosied commented on dev/wlp-jakartaee-transform/rules/jakarta-versions.properties:
```diff
@@ -1,4 +1,3 @@
-com.ibm.wsspi.el=[4.0,5)
```
Transformations are not longer necessary as this package is now used internally. (Except when building com.ibm.ws.cdi.web)

volosied commented on dev/com.ibm.ws.jsp/bnd.bnd:
```diff
@@ -61,7 +61,8 @@ Export-Package: com.ibm.ws.jsp.runtime;version=2.2;thread-context=true, \
    org.apache.jasper.compiler;thread-context=true;version=2.2, \
-   org.apache.jasper.runtime;thread-context=true;version=2.2
+   org.apache.jasper.runtime;thread-context=true;version=2.2, \
+   com.ibm.wsspi.el
```
Mostly used internally within com.ibm.ws.jsp.

Otherwise, this package is exported to build the com.ibm.ws.cdi.web bundle. 

https://github.com/OpenLiberty/open-liberty/blob/2fd4a880754c37a988c5ed9ac4f1ea5988e465d6/dev/com.ibm.ws.cdi.web/src/com/ibm/ws/cdi/web/impl/el/WrappedELExpressionFactory.java#L28

volosied commented:
#build

volosied commented:
Hoped it wouldn't happen, but it did -- `com.ibm.ws.openwebbeans.impl` failed to compile. Will need to update CL

volosied commented:
Build identified and error when cdi & servlet are enabled.

```
[ERROR   ] CWWKF0029E: Could not resolve module: com.ibm.ws.cdi.web.jakarta [123]
  Unresolved requirement: Import-Package: com.ibm.wsspi.el; version="[2.2.0,3.0.0)"
```

io.openliberty.cdi3.0-servlet5.0 only enables the private jsp feature 
https://github.com/OpenLiberty/open-liberty/blob/2fd4a880754c37a988c5ed9ac4f1ea5988e465d6/dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.cdi3.0-servlet5.0.feature#L8


https://github.com/OpenLiberty/open-liberty/blob/cd9f0aec1ef34d4809ab4f5c0420ecd6dd4c4252/dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.jakarta.pages-3.0.feature#L4-L7


Explains why  com.ibm.wsspi.el; cannot be found. 

Without changing the feature files, the only bundles to place this package are :
Servlet API
EL API
JSP API
(3 are taken from the code snippet above since io.openliberty.cdi3.0-servlet5.0 enables /io.openliberty.jakarta.pages-3.0 ) 

None seem good candidates? I don't want to create a new bundle as that would slow start up just slightly and it seems unnecessary to had a bundle for a single class.

volosied committed:
Create io.openliberty.el.internal.cdi

Change references to the new ELFactoryWrapperForCDI package

Update features

Fix more ELFactoryWrapperForCDI references

volosied committed:
Corrections & Address Review Comments (Mostly Copyright Updates)

volosied commented:
Personal Build Passed -- https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_13o2kADuEeyB1uNHimT5CA&tab=com.ibm.team.build.web.ui.internal.editors.result.properties.PropertiesProvider

(Needed to use an CL Branch)

pnicolucci commented on dev/com.ibm.ws.jsp.2.3/bnd.bnd:
```diff
@@ -107,4 +107,5 @@ instrument.classesExcludes: org/apache/**/*.class
 	com.ibm.ws.org.osgi.annotation.versioning;version=latest, \
-	com.ibm.ws.javaee.version;version=latest
+	com.ibm.ws.javaee.version;version=latest, \
+   io.openliberty.el.internal.cdi;version=latest
```
Can you fix the indent here?

volosied commented:
#run-libby-bot
